src/ui/validate_transaction_nbgl.c

```diff
@@ -151,107 +203,60 @@ static void continue_review(void) {
-            p3 = REVIEW_P3_SWAP;
-            p4 = REVIEW_P4_SWAP;
-            break;
-        case SELL:
-        case SELL_NG:
-            dyn_string_1 = G_swap_ctx.sell_transaction.in_currency;
-            dyn_string_2 = G_swap_ctx.sell_transaction.out_currency;
-            p3 = REVIEW_P3_SELL;
-            p4 = REVIEW_P4_SELL;
-            break;
-        case FUND:
-        case FUND_NG:
-            dyn_string_1 = "";
-            dyn_string_2 = G_swap_ctx.fund_transaction.in_currency;
-            p3 = REVIEW_P3_FUND;
-            p4 = REVIEW_P4_FUND;
-            break;
-    }
+/*********************
+ *     CALLBACKS     *
+ *********************/
 
-    // Detect if we should display on 2 or 3 lines.
-    if ((G_swap_ctx.subcommand == FUND || G_swap_ctx.subcommand == FUND_NG) ||
-        (strlen(dyn_string_1) + strlen(dyn_string_2) >= 10)) {
-        PRINTF("Review title and confirm on 3 lines\n");
-        // Move the "to" to the second line with the operation
-        delimitor_1 = '\n';
-        delimitor_2 = ' ';
-        delimitor_3 = '\n';
-    }
+static void accept_tx(void) {
+    nbgl_useCaseSpinner("Processing");
+    reply_success();
+    G_swap_ctx.state = WAITING_SIGNING;
+}
 
-    snprintf(review_title,
-             sizeof(review_title),
-             "%s%c%s%c%s%s%c%s%s",
-             REVIEW_P1_TITLE,
-             delimitor_1,
-             REVIEW_P2,
-             delimitor_2,
-             p3,
-             dyn_string_1,
-             delimitor_3,
-             p4,
-             dyn_string_2);
+static void reject_tx(void) {
+    PRINTF("User refused transaction\n");
+    reply_error(USER_REFUSED);
+    nbgl_useCaseReviewStatus(STATUS_TYPE_TRANSACTION_REJECTED, ui_idle);
+}
 
-    snprintf(review_confirm,
-             sizeof(review_confirm),
-             "%s%c%s%c%s%s%c%s%s%s",
-             REVIEW_P1_CONFIRM,
-             delimitor_1,
-             REVIEW_P2,
-             delimitor_2,
-             p3,
-             dyn_string_1,
-             delimitor_3,
-             p4,
-             dyn_string_2,
-             REVIEW_P5_CONFIRM);
-
-    nbgl_useCaseReviewStart(&C_icon_exchange_64x64,
-                            review_title,
-                            NULL,
-                            "Reject transaction",
-                            continue_review,
-                            rejectUseCaseChoice);
+// If the user asks for message rejection, ask for confirmation
+static void rejectUseCaseChoice(void) {
+    nbgl_useCaseConfirm("Reject transaction?",
+                        NULL,
+                        "Yes, reject",
+                        "Go back to transaction",
+                        reject_tx);
 }
 
-void display_signing_success(void) {
-    nbgl_useCaseStatus("TRANSACTION\nSIGNED", true, ui_idle);
+static void review_choice(bool confirm) {
+    if (confirm) {
+        accept_tx();
+    } else {
+        rejectUseCaseChoice();
```

Comment:
here we could directly call the functions called in `reject_tx` and  then remove `rejectUseCaseChoice `and `reject_tx`

Reply:
Done